text/0000-buildpacks-config.md

```diff
@@ -0,0 +1,314 @@
+
+ - **Application API**: A new specification detail the contract and schema between the application, it's configuration, and the `lifecycle` components.
+ - **Platform API**: Existing specification detailing contract between the `platform` and `lifecycle`.
+ - **platform config**: A new file that provides a Platform API specific configuration to the platform from an Application API configuration.
+ - **project.toml (Project Descriptor)**: An existing configuration file created by this project with higher asperations of configuring not just buildpacks but other tools and platforms. See "Supersedes" meta issues above.
+
+# Motivation
+[motivation]: #motivation
+
+The current use of the _project descriptor_ has a few issues:
+
+### Cross-Platform Support
+
+The current specification treats the _project descriptor_ as an "extension" specification meaning that it may only be optionally supported by platforms to be compliant.
+
+From a user's point-of-view, it's unexpected to have their application built one way locally via `pack` (which supports `project.toml`) and another way using another platform such as `Tekton` (which doesn't support `project.toml`) where both platforms natively support Cloud Native Buildpacks. See reported [issue][issue-tekton-33].
+
+### Obscure
+
+Due to the generic intent of `project.toml`, it is not inherently clear that a project is configured to be built by Cloud Native Buildpacks using a `project.toml`. Compare this to recognizing a platform or system specific configuration file such as `.github/`, `Dockerfile`, `travis.yml`, etc.
+
+```text
+.
+├── .git/
+├── .github/
+├── src/
+├── .gitignore
+├── .gitpod.yml
+├── README.md
+├── codecov.yml
+├── golangci.yaml
+├── Dockerfile
+└── project.toml
+```
+
+### Complex Syntax
+
+In trying to support multiple use cases in a single file, the syntax become slightly more complex by the use of ["reverse domain namespacing"][reverse-domain-namespacing].
+
+```toml
+[_]
+api = "0.2"
+id = "<string>" # machine readable
+name = "<string>" # human readable
+version = "<string>"
+authors = ["<string>"]
+documentation-url = "<url>"
+source-url = "<url>"
+
+[[_.licenses]]
+type = "<string>"
+uri = "<uri>"
+
+[io.buildpacks]
+api = "0.1"
+
+[io.buildpacks.build]
+include = ["<string>"]
+exclude = ["<string>"]
+
+[[io.buildpacks.build.buildpacks]]
+id = "<string>"
+version = "<string>"
+uri = "<string>"
+
+[[io.buildpacks.build.env]]
+name = "<string>"
+value = "<string>"
+```
+
+[reverse-domain-namespacing]: https://github.com/buildpacks/rfcs/blob/main/text/0084-project-descriptor-domains.md
+[issue-tekton-33]: https://github.com/buildpacks/tekton-integration/issues/33
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposed solution to the issues listed in the [motivation](#motivation) is to create a **Cloud Native Buildpacks specific file** that **MUST be supported by platforms**.
+
+## File
+
+`buildpacks.<ext>` where `<ext>` corrolates to a supported format.
```

Comment:
this is way too easily confused with `buildpack.toml`

Reply:
I agree but do we expect app developers to be using `buildpack.toml`? Alternatively, maybe something like `Buildpacks`, with no extension, similar to `Dockerfile`.

Reply:
> I agree but do we expect app developers to be using buildpack.toml?

If we don't, I think that makes the problem worse, because they won't even notice when they need to pay attention to the differences.